File: wit-bindings/cpp/test-module/src/module.cpp
#include "../../bindings/orbital_cpp.h"
#include <iostream>

namespace exports {
namespace orbital {
namespace core {
namespace module {
std::optional<commands::CommandBuffer> Startup() {
  std::cout << "'Hello World!' from WIT TestModule: C++" << std::endl;

  // Return Rust's "None"
  return std::nullopt;

  /* To return Rust's "Some":
   * commands::CommandBuffer buffer;
   * commands::Command cmd;
   * cmd.variants = commands::Command::RegisterSystem { wit::string("MySystem")
   * }; buffer.commands.push_back(std::move(cmd)); return buffer;
   */
}

} // namespace module
} // namespace core
} // namespace orbital
} // namespace exports

File: wit-bindings/csharp/test-module/Module.cs
using OrbitalWorld.wit.Exports.orbital.core.v0_1_0;

public partial class ModuleExportsImpl {
  public static ICommandsExports.CommandBuffer? Startup() {
    Console.WriteLine("'Hello World!' from WIT TestModule: C#");

    // // Create a sample command buffer with a register-system command
    // var commands = new
    // System.Collections.Generic.List<ICommandsExports.Command>();
    // commands.Add(ICommandsExports.Command.RegisterSystem("test-system"));
    //
    // var commandBuffer = new ICommandsExports.CommandBuffer(commands);
    // return commandBuffer;

    return null;
  }
}
